sdk/src/main/java/com/kin/ecosystem/Kin.java

```diff
@@ -74,27 +84,106 @@ private static Kin getInstance() {
 		return instance;
 	}
 
-	public static void start(@NonNull Context appContext, @NonNull WhitelistData whitelistData,
-		@NonNull KinEnvironment environment)
-		throws ClientException, BlockchainException {
+	synchronized static void initialize(Context appContext) throws ClientException {
```

Comment:
does it need to be synchronized?

Reply:
yes to be on the safe side.

Reply:
it is not safe - we can create ANR - we can only call it from this package and we are only calling it from KinEcosystemInitProvider. using synchronized have some performance cost and we shouldn't use it when called from Provider. There is another issue that I just thought about now (sorry it just now)- what if the developer wants to test the SDK on a small set of its user? if we place init in KinEcosystemInitProvider this code will run on each instance whether or not it is part of the AB.

Reply:
I also agree, and this is a use case I didn't even think about, but generally, this kind of SDK, which has nothing special to manage across the entire application lifecycle, is not in a position to force itself whenever you take it. I think that developer should choose whenever he wants to init it. (I'm also not a fan of generally expose it as static from similar reasons)